hibernate-core/src/main/java/org/hibernate/sql/ast/spi/AbstractSqlAstTranslator.java

```diff
@@ -1269,20 +1269,15 @@ protected void visitInsertStatementOnly(InsertSelectStatement statement) {
 		boolean firstPass = true;
 
 		final List<ColumnReference> targetColumnReferences = statement.getTargetColumns();
-		if ( targetColumnReferences == null ) {
```

Comment:
```
	@Override
	public List<ColumnReference> getTargetColumns() {
		return targetColumnReferences == null ? Collections.emptyList() : targetColumnReferences;
	}
```
as per the above logic, `statement.getTargetColumns()` is non-null.

On the other hand, checking emptiness seems unnecessary as well for either HQL or Criteria has guarded against such possibility (HQL's BNF grammar doesn't allow empty insert columns list; Criteria would have throws `SemanticException`).